core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -602,7 +602,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
                                       maxMessageSize = config.messageMaxBytes,
                                       maxIoBytesPerSecond = config.logCleanerIoMaxBytesPerSecond,
                                       backOffMs = config.logCleanerBackoffMs,
-                                      enableCleaner = config.logCleanerEnable)
+                                      enableCleaner = config.logCleanerEnable,
+                                      brokerCompressionType = config.compressionType)
```

Comment:
You are calling `config.compressionType`. `config` in `KafkaServer` is `val config: KafkaConfig`? The topic configs are under `configs`. See how `LogManager` is created:

```scala
new LogManager(logDirs = config.logDirs.map(new File(_)).toArray,
                     topicConfigs = configs,
                     defaultConfig = defaultLogConfig,
```

It is my turn to ask, am I missing something?